R/formula.R

```diff
@@ -0,0 +1,422 @@
+library(stringr)
+
+extract_terms_response <- function(formula) {
+  tt <- terms(formula)
+  terms <- attr(tt, "term.labels")
+  variables <- as.list(attr(tt, "variables")[-1])
+  response <- attr(tt, "response")
+  global_intercept <- attr(tt, "intercept") == 1
+
+  hier <- str_match(terms, "\\|")
```

Comment:
Agreed. Many of the functions in this file came to life to handle formulas in the most general way that I was able to. There are so many different ways of specifying a formula that I'm sure this can break in trizillion ways, so this is something to keep in mind to make our management more robust in the future. As users come up with weird and complex formulas it will be easier to improve.